quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -78,10 +232,178 @@ impl Builder {
         self.queue_capacity = capacity;
         self
     }
+
+    /// Gives the default sender relevant connection info
+    pub fn with_connection_info(mut self, connection_info: ConnectionInfo) -> Self {
```

Comment:
This should just take a reference and take the information it needs. We usually try to keep these structs as `!Copy` for future flexibility.

```suggestion
    pub fn with_connection_info(mut self, connection_info: &ConnectionInfo) -> Self {
```